specification/billing/resource-manager/Microsoft.Billing/stable/2024-04-01/billingAccount.json

```diff
@@ -0,0 +1,1273 @@
+              "Location": {
+                "type": "string"
+              },
+              "Retry-After": {
+                "type": "integer",
+                "format": "int32"
+              }
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-examples": {
+          "BillingAccountsUpdate": {
+            "$ref": "./examples/billingAccountsUpdate.json"
+          },
+          "BillingAccountUpdateWithPONumber": {
+            "$ref": "./examples/billingAccountUpdateWithPONumber.json"
+          }
+        }
+      }
+    },
+    "/providers/Microsoft.Billing/billingAccounts": {
+      "get": {
+        "tags": [
+          "BillingAccount"
+        ],
+        "description": "Lists the billing accounts that a user has access to.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/billing/"
+        },
+        "operationId": "BillingAccounts_List",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "in": "query",
+            "name": "includeAll",
+            "description": "When true, results will include Billing Accounts that the user does not have a direct role assignment on if the user has one of the following AAD roles: Global Administrator, Global Reader, Billing Administrator.",
+            "type": "boolean",
+            "default": false
+          },
+          {
+            "in": "query",
+            "name": "includeAllWithoutBillingProfiles",
+            "description": "When true, results will include Billing Accounts that are not fully created if the user has one of the following AAD roles: Global Administrator, Global Reader, Billing Administrator.",
+            "type": "boolean",
+            "default": false
+          },
+          {
+            "in": "query",
+            "name": "includeDeleted",
+            "description": "When true, results will include any billing accounts in a deleted state.",
+            "type": "boolean",
+            "default": false
+          },
+          {
+            "in": "query",
+            "name": "includePendingAgreement",
+            "description": "Includes billing accounts with agreement pending signature that the user has access to.",
+            "type": "boolean",
+            "default": false
+          },
+          {
+            "in": "query",
+            "name": "includeResellee",
+            "description": "Includes the customer's billing account of Microsoft Partner Agreement that the user has access to.",
+            "type": "boolean",
+            "default": false
+          },
```

Comment:
I somehow find the idea of this option a little worrying... will you be ensuring that it can't be abused in combination with things like 'includeAll' to see all accounts in another tenant?

Reply:
Yes, no matter what flags are used, the user only sees accounts that they have access to.

IncludeAll really means all accounts no matter what status because the day-one behavior of this API was to hide deleted billing accounts..

Reply:
Is this limited with OBO?

What else do I find worrying about it.... that it breaks assumptions, about the tenants of the customer or the request matching those of the response... who could be relying on such assumptions? Hmm.

Reply:
>IncludeAll really means all accounts no matter what status because the day-one behavior of this API was to hide deleted billing accounts..

That's odd, I thought you had a separate flag for showing deleted accounts:

`includeDeleted`

Reply:
Ah right, this is different. Global admins, Global Readers and Billing Admins are able to see all billing accounts in their tenant. Global admins can also self-elevate in CPAS to perform write actions on a billing account. But that's a more expensive operation to show all accounts in those minority cases so the includeAll flag is makes the extra calls under the hood to show the accounts for those with the AAD role.

Reply:
Does this rely on custom auth, as opposed to standard RBAC?

Reply:
Yes, most of BillingRP does. There is a long-pull effort to migrate to PAS.

Reply:
@bradenwatkins I suspect this API involving data from multiple tenants looks could present challenges for migrating to PAS. I'm curious whether data from multiple tenants really _needs_ to be fetched in a single request this way. Designing it this way seems like it might also add unnecessary complexity to the implementation of your authorization.

Reply:
@ramkd12 can you comment here on the design and behavior of the BillingAccounts API and respond to Tim's concern?

Reply:
There are very complex relationships for some companies like Walmart which needs multi tenant discovery of billing accounts across their subsidiaries. Also they operate on PUID based discovery. 

There is a long list of items which PAS doesn't support for billing scenarios which is why billing ahs its own Auth and yes, Multi tenancy is just one of them. There are active streams of work happening between various teams of Commerce and PAS and ARG for convergence but that is a long term plan.

Also, just to add, none of these are newly added in this version. These all are shipped years back and this version is carrying forward to meet their live scenarios.